### PR TITLE
Port cats effect 3

### PR DESCRIPTION
**Pull request by cquiroz**
This is a partial port to CE3. While all test are passing there are some uncompleted bits which I'll comment inline

### CODE REVIEW

File: js/src/main/scala/crystal/react/ContextProvider.scala [hunk 1]
```diff
@@ -12,17 +11,13 @@ object ContextProvider {
...
-      cs:             ContextShift[F]
-    ): StateComponent[C] =
+    def apply[C](ctx:        Ctx[F, C], initCtx:    C)(
+      render:                VdomNode
+    )(implicit reusabilityC: Reusability[C], async: Async[F]): StateComponent[C] =
       ScalaComponent
         .builder[Unit]
         .initialState(initCtx)
-        .render($ => ctx.provide(ViewF.fromState[F]($))(render))
+        .render($ => ctx.provide(ViewF.fromState[F](async)($))(render))
```
Comment (cquiroz) [inline js/src/main/scala/crystal/react/ContextProvider.scala:20]: `async` should pass implicitly but for some reason it doesn't and I have to pass it explicitly

File: js/src/main/scala/crystal/react/StreamRendererBackend.scala [hunk 2]
```diff
@@ -2,39 +2,44 @@ package crystal.react
...
 import cats.effect._
 import cats.syntax.all._
 import japgolly.scalajs.react._
 import org.typelevel.log4cats.Logger
+import cats.effect.std.Dispatcher
+import scala.concurrent.Future
 
-abstract class StreamRendererBackend[F[_]: ConcurrentEffect: Logger, A](stream: fs2.Stream[F, A]) {
-  private var cancelToken: Option[CancelToken[F]] = None
+abstract class StreamRendererBackend[F[_]: Async: Logger, A](stream: fs2.Stream[F, A]) {
```
Comment (cquiroz) [inline js/src/main/scala/crystal/react/StreamRendererBackend.scala:12]: This needs work, the start/stop methods aren't doing anything at the moment

File: js/src/main/scala/crystal/react/StreamRendererBackend.scala [hunk 3]
```diff
@@ -2,39 +2,44 @@ package crystal.react
...
-    )
+  // TODO Should be private
+  lazy val evalCancellable: F[CancelToken] =
+    Dispatcher[F]
+      .use { dispatcher =>
+        val fa: F[Unit] = stream
+          .evalMap(v => streamSetState(v.ready))
+          .compile
+          .drain
+        val ct          = dispatcher.unsafeRunCancelable(fa)
```
Comment (cquiroz) [inline js/src/main/scala/crystal/react/StreamRendererBackend.scala:28]: I'm fairly sure this isn't right

File: shared/src/main/scala/crystal/viewF.scala [hunk 4]
```diff
@@ -26,21 +24,22 @@ sealed trait ViewOps[F[_], G[_], A] {
...
+final class ViewF[F[_]: Async, A](val get: A, val mod: (A => A) => F[Unit])
     extends ViewOps[F, Id, A] {
 
   val set: A => F[Unit] = a => mod(_ => a)
 
   // In a ViewF, we can derive modAndGet. In ViewOptF and ViewListF we have to pass it, since their
   // mod functions have no idea of the enclosing structure.
-  val modAndGet: (A => A) => F[A] = f =>
+  val modAndGet: (A => A) => F[A] = f => {
     Sync[F].delay(Promise[A]()).flatMap { p =>
```
Comment (cquiroz) [inline shared/src/main/scala/crystal/viewF.scala:35]: I tried to do this using `Deferrable` but couldn't

Comment (rpiaggio) [reply shared/src/main/scala/crystal/viewF.scala:35]: Neither could I.

**Pull request merged**